### ISSUE TITLE
Change RomCommand op and funcName attributes from string to enumerated for performance reasons

### ISSUE THREAD

**Issue by fractasy** [0xPolygonHermez/zkevm-prover#17]
- 2 enumerated types must be crated, one for each element, e.g. tOp, tFuncName
- First enum value will be empty=0
- 2 conversion functions must be created:
    - tOp string2op (string &s)
    - tFuncName string2funcname (string &s)
- eval command code must be changed to use switch instead of string comparisons

**Comment by fractasy** [comment]
Example:
typedef enum
{
    empty = 0,
    getGlobalHash = 1,
    ...
} tFunction;

tFunction string2function(string &s)
{
    if (s=="getGlobalHash") return getGlobalHash;
    ...
    cerr << "Error: string2function() invalid string=" << s << endl;
    exit(-1);
}

**Comment by agnusmor** [comment]
tFunction and tOp enum types has been created. 

Also we have created functions to convert from this enums to the string identifier (and viceversa):

tFunction string2Function(string s);
tOp string2Op(string s);
string function2String(tFunction f);
string op2String(tOp op);

When loading de ROM we convert the string identifier for the funcName and op properties to his corresponding enum value. This allows us to use a switch expression in the eval_functionCall and evalCommand functions, instead to use if/else string comparation and therefore improve the performance of this functions.

We have ran some performance test to compare "switch" vs "if/else" and when using "switch" is around 2,5 times faster than "if/else" (in average).

Some metrics of the performance tests (for eval_functionCall function):

TEST "if/else"

With "beforeLast" funcName at the beginnig of the evaluations:
Eval time: 322.705 ms, called 2089943 times, so 0 us/time
Eval time: 328.522 ms, called 2089943 times, so 0 us/time
Eval time: 337.55 ms, called 2089943 times, so 0 us/time
Eval time: 369.786 ms, called 2089943 times, so 0 us/time
Eval time: 340.121 ms, called 2089943 times, so 0 us/time
Eval time: 340.528 ms, called 2089943 times, so 0 us/time
Eval time: 341.104 ms, called 2089943 times, so 0 us/time
Eval time: 348.985 ms, called 2089943 times, so 0 us/time
AVG=341,162 ms

With "beforeLast" funcName at the end of the evaluations:
Eval time: 1362.81 ms, called 2089943 times, so 0 us/time
Eval time: 1274.71 ms, called 2089943 times, so 0 us/time
Eval time: 1287.56 ms, called 2089943 times, so 0 us/time
Eval time: 1265.17 ms, called 2089943 times, so 0 us/time
Eval time: 1312.11 ms, called 2089943 times, so 0 us/time
Eval time: 1294.51 ms, called 2089943 times, so 0 us/time
Eval time: 1277.56 ms, called 2089943 times, so 0 us/time
Eval time: 1269.87 ms, called 2089943 times, so 0 us/time
AVG=1293,03 ms

TEST "swtich"
With "beforeLast" funcName at the beginnig of the evaluations:
Eval time: 305.027 ms, called 2089943 times, so 0 us/time
Eval time: 295.691 ms, called 2089943 times, so 0 us/time
Eval time: 350.34 ms, called 2089943 times, so 0 us/time
Eval time: 326.981 ms, called 2089943 times, so 0 us/time
Eval time: 321.467 ms, called 2089943 times, so 0 us/time
Eval time: 308.556 ms, called 2089943 times, so 0 us/time
Eval time: 311.367 ms, called 2089943 times, so 0 us/time
Eval time: 315.516 ms, called 2089943 times, so 0 us/time
AVG=316,868 ms

With "beforeLast" funcName at the end of the evaluations:
Eval time: 300.855 ms, called 2089943 times, so 0 us/time
Eval time: 287.03 ms, called 2089943 times, so 0 us/time
Eval time: 289.316 ms, called 2089943 times, so 0 us/time
Eval time: 301.455 ms, called 2089943 times, so 0 us/time
Eval time: 297.036 ms, called 2089943 times, so 0 us/time
Eval time: 311.216 ms, called 2089943 times, so 0 us/time
Eval time: 307.034 ms, called 2089943 times, so 0 us/time
Eval time: 326.656 ms, called 2089943 times, so 0 us/time
AVG=302,574 ms
